### PR TITLE
Switch from optparse to argparse (#318)

### PR DESCRIPTION
**Pull request by whimboo**
This PR will fix issue #318.

### CODE REVIEW

**Comment by whimboo** [comment]
This PR depends on PR #317. So only the 2nd commit should be obeyed for now.

File: mozdownload/cli.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+                       dest='changeset',
+                       help='Changeset of the try build to download')
+
+    args = parser.parse_args()
+
+    # Gives instructions to user when no arguments were passed
+    if len(sys.argv) == 1:
+        print __doc__
+        parser.format_usage()
+        print 'Specify --help for more information on args. ' \
```
Comment (AutomatedTester) [inline mozdownload/cli.py]: unneeded `\`

**Comment by AutomatedTester** [comment]
r+ when comment is addressed

File: mozdownload/cli.py (hunk 1 above)
Comment (whimboo) [inline mozdownload/cli.py]: Well, without using rounded brackets for the print statement the next line would have to start at the same column without indentation. It's kinda hard to read. So I will change the file to use `print('...')` instead.

**Pull request merged**